tests/run-make/msvc-wholearchive/rmake.rs

```diff
@@ -0,0 +1,52 @@
+//! This is a regression test for #129020
+//! It ensures we can use `/WHOLEARCHIVE` to link a rust staticlib into DLL
+//! using the MSVC linker
+
+//@ only-msvc
```

Comment:
Would be nice to also cover `windows-gnullvm` that has the same issue.

Reply:
I've not tried gnullvm specifically but the llvm linker has another issue which I think needs solving on their end first (which is why even the msvc test skips lld-link).

```
.idata$4 should not refer to special section 0
```

LLVM seems to not properly recognise `.idata$4` and `.idata$5` as special unless they're specifically in short import libraries. incidentally this also makes it break on MSVC long import libraries. Though since they haven't been in use since the 90's I guess there's not been much pressure to support it.

Reply:
We probably should open a new issue to track this case?

Reply:
Maybe? How do we usually track upstream issues?

Reply:
We have a special label for waiting on LLVM: `S-waiting-for-llvm` for describing the dragon is eepy

Reply:
> Though since they haven't been in use since the 90's I guess there's not been much pressure to support it.

ld.bfd used by `windows-gnu` still uses them as it still cannot output short import libraries 😉